### ISSUE TITLE
Hot Updates Reporting Errors Causing Project Crashes

### ISSUE THREAD

**Issue by liruifengv** [withastro/docs#6406]
### 📝 Issue Description

I am translating the docs. 

Update, save, hot update, open the page again and find it crashes. The following errors were reported:

![image](https://github.com/withastro/docs/assets/25167721/5f33451e-4254-4d06-9409-5c9c5489bec1)

![image](https://github.com/withastro/docs/assets/25167721/4edf608c-629d-44b0-a8ee-8014047cd179)
![image](https://github.com/withastro/docs/assets/25167721/18de08e9-5012-4d00-9ba0-0d8918611f56)

![image](https://github.com/withastro/docs/assets/25167721/209f9b7a-b61f-4238-ae86-7b5bbe067ba9)

### 📋 On which page(s) it occurs

/zh-cn/reference/api-reference/

maybe all page. Not just one page.

### 🤔 Expected Behavior

Development servers don't crash

### 👀 Current Behavior

Development servers crashed.

### 🖥️ Browser

Chrome

### 📄 Additional Information

windows 11. Powershell.

**Comment by delucis** [comment]
Thanks for the issue @liruifengv! We were aware of this one as it’s something @sarah11918 runs into quite often too (although curiously, I never have, so I guess it depends on something system specific). There’s some more information about the cause from @matthewp [on Discord](https://discord.com/channels/830184174198718474/1070481941863878697/1183854162203656213):

> I tracked down the errors you some times see in Starlight apps (such as Astro Docs) that look something like this: `Cannot read properties of null (reading 'EditLink')`
>
> tldr; this is a Vite bug related to circular dependencies. When modules depend on each other and an update occurs circular dependencies can be in an invalid state by the time they are re-executed, causing this error.

Assuming the underlying Vite bug cannot be fixed, the solution in Starlight is to restructure how we currently organise our virtual component module to avoid circular dependencies.

Currently, we have a single `virtual:starlight/components` module which exports all the Starlight layout components. But some of these components themselves import from the same module causing the circular chain, e.g. `Footer` imports `EditLink` from the virtual module and is then itself exported from the virtual module.

I think if we refactor to use a separate module for each component, we would avoid this error, i.e. usage would become:
```diff
- import { EditLink } from 'virtual:starlight/components';
+ import EditLink from 'virtual:starlight/components/EditLink';
```

Will transfer this issue to the Starlight repo as it needs resolving there.
